official/transformer/transformer_main.py

```diff
@@ -353,12 +360,12 @@ def run_loop(
 def define_transformer_flags():
   """Add flags and flag validators for running transformer_main."""
   # Add common flags (data_dir, model_dir, train_epochs, etc.).
-  flags_core.define_base(multi_gpu=False, num_gpu=False)
+  flags_core.define_base(multi_gpu=False)
```

Comment:
I don't think we need the multi_gpu=False here, and it is deceptive, so better to remove?

Reply:
Perhaps we should remove the `multi_gpu` flag, and use `num_gpu` instead? MNIST uses `multi_gpu` but not `num_gpu`, and the resnet models use `num_gpu` but not `multi_gpu`.

I have a question/suggestion about the `num_gpu` flag (+@robieta about this) - Currently, the default value of `num_gpu` is 0 or 1 depending on if there is a GPU. Can we change this default to None instead?
OneDeviceStrategy requires more memory than having no DistributionStrategy (batch size of 4096 doesn't causes OOM errors when using OneDeviceStrategy). When the user doesn't specify num_gpus, we should default to using no DistributionStrategy.

Reply:
Hm. Can you ping DistStrat and see if this is expected, and what the advantages of using OneDevice are/will be?

Reply:
Hmm, tried running it again, and it went through without failing. Seems like this might be Transformer getting unlucky with the dynamic batching? For now, I'll leave it as is. 

What are your thoughts about the `num_gpu` vs `multi_gpu` flags?

Reply:
We want to get everything on `num_gpus` and remove `multi_gpu` altogether.